snap/container.go

```diff
@@ -351,33 +353,48 @@ func validateContainer(c Container, needsrx, needsx, needsr, needsf, noskipd map
 
 		if mode.IsDir() {
 			if mode.Perm()&0555 != 0555 {
-				logf("in %s %q: %q should be world-readable and executable, and isn't: %s", contType, name, path, mode)
-				hasBadModes = true
+				err := fmt.Errorf("%q should be world-readable and executable, and isn't: %s", path, mode)
+				logf("in %s %q: %v", contType, name, err)
+				if firstBadModeErr == nil {
```

Comment:
This works. 
It made me think about error aggregation and identification of first instance as a generally useful concept.

Could create error type for aggregation, containing list of `Errors []error` that defaults to error[0] for message, but can be iterated to concert errors to e.g. json output or detail log entry etc as required.
This could the become e.g. accErrorBadModes.add(...)

Anyways, more applicable to longer term solution...